lib/actionview_precompiler/render_parser.rb

```diff
@@ -117,7 +117,24 @@ def parse_render_from_options(options_hash)
       end
 
       render_type = (keys & render_type_keys)[0]
-      template = parse_str(options_hash[render_type])
+
+      node = options_hash[render_type]
+      if node.string?
+        template = node.to_string
+      else
+        if node.variable_reference?
+          dependency = node.variable_name.sub(/\A(?:\$|@{1,2})/, "")
+        elsif node.vcall?
+          dependency = node.variable_name
+        elsif node.call?
+          dependency = node.call_method_name
+        else
+          return
+        end
+        object_template = true
+        template = "#{name.pluralize}/#{name.singularize}"
```

Comment:
@jhawthorn how do you feel about this method of reducing the duplication? I don't love the nested `if` statements, but I think the code overall reads clearer.

Reply:
Yeah. Seems good. The `return` is a bit awkward at that nesting, but I don't have a better alternative